From 40db061600f4b550e313ef72325a1711fae9ad1d Mon Sep 17 00:00:00 2001
From: Jinesh K. Jayakumar <quic_jineshk@quicinc.com>
Date: Mon, 18 Sep 2023 13:57:37 -0400
Subject: [PATCH] macsec_linux: Set predictable macsec interface name

Generate predictable interface name for macsec using physical
interface name and macsec port number.

Signed-off-by: Jinesh K. Jayakumar <quic_jineshk@quicinc.com>
---
 src/drivers/driver_macsec_linux.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/drivers/driver_macsec_linux.c b/src/drivers/driver_macsec_linux.c
index 3dba13ce7..22e266ccc 100644
--- a/src/drivers/driver_macsec_linux.c
+++ b/src/drivers/driver_macsec_linux.c
@@ -1082,6 +1082,19 @@ static struct rtnl_link * lookup_sc(struct nl_cache *cache, int parent, u64 sci)
 }
 
 
+/* Set macsec interface name to <phys_intf>s<sci_port> */
+static int set_iface_name(struct transmit_sc *sc,
+		struct macsec_drv_data *drv,
+		struct rtnl_link *link)
+{
+	char ifname[IFNAMSIZ + 1];
+
+	os_snprintf(ifname, sizeof(ifname), "%ss%u", drv->common.ifname, be_to_host16(sc->sci.port));
+
+	rtnl_link_set_name(link, ifname);
+}
+
+
 /**
  * macsec_drv_create_transmit_sc - Create secure connection for transmit
  * @priv: private driver interface data from init()
@@ -1115,6 +1128,7 @@ static int macsec_drv_create_transmit_sc(
 		return -1;
 	}
 
+	set_iface_name(sc, drv, link);
 	rtnl_link_set_link(link, drv->parent_ifi);
 
 	sci = mka_sci_u64(&sc->sci);
-- 
2.34.1
